### PR TITLE
Added proxy support

### PR DESCRIPTION
**Pull request by Ferrerohub**

### CODE REVIEW

**Comment by Ferrerohub** [comment]
OMG, I'm using the [Ray Wenderlich's text-style recommendations](https://github.com/raywenderlich/swift-style-guide#correctness), but code doesn't look the same as here

File: Sources/Telegrammer/Network/Bot.Settings+Proxy.swift [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+import Foundation
+import AsyncHTTPClient
+
+public extension Bot.Settings {
+  struct Proxy {
+    public var host: String
+    public var port: Int
+    public var authorization: Authorization?
+    
+    public static func server(host: String, port: Int) -> Proxy {
```
Comment (givip) [inline Sources/Telegrammer/Network/Bot.Settings+Proxy.swift]: This method is redundant, its covered by following method

File: Sources/Telegrammer/Bot/Telegram/Bot.swift [hunk 2]
```diff
@@ -49,11 +50,13 @@ public final class Bot: BotProtocol {
         self.settings = settings
         self.clientWorker = MultiThreadedEventLoopGroup(numberOfThreads: numThreads)
         let groupProvider = HTTPClient.EventLoopGroupProvider.shared(self.clientWorker)
+        let proxy = settings.proxy.map(HTTPClient.Configuration.Proxy.init)
```
Comment (givip) [inline Sources/Telegrammer/Bot/Telegram/Bot.swift:53]: Interesting approach, looks like I missed this from Optional macOS 10.15 updates

**Review by givip — COMMENTED** [review]
Yes, you are right, code style is not as described by Ray Wenderlich. Have to fix CONTRIBUTING guide

File: Sources/Telegrammer/Bot/Telegram/Bot.swift [hunk 3]
```diff
@@ -25,6 +25,7 @@ public final class Bot: BotProtocol {
         public var serverHost: String = "api.telegram.org"
         public var serverPort: Int = 443
         public var webhooksConfig: Webhooks.Config?
+        public var proxy: Bot.Settings.Proxy? = nil
```
Comment (givip) [inline Sources/Telegrammer/Bot/Telegram/Bot.swift]: no `nil` needed

File: Sources/Telegrammer/Network/Bot.Settings+Proxy.swift [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+//  Bot.Settings+Proxy.swift
+//  AsyncHTTPClient
+//
+//  Created by Ярослав Попов on 07.01.2020.
+//
+
+import Foundation
+import AsyncHTTPClient
+
+public extension Bot.Settings {
```
Comment (givip) [inline Sources/Telegrammer/Network/Bot.Settings+Proxy.swift:11]: Please make 4 spaces everywhere

File: Sources/Telegrammer/Network/BotClient.swift [hunk 5]
```diff
@@ -22,15 +22,15 @@ public class BotClient {
...
     ///   - worker: Worker on which will be performed request
+    ///   - proxy: Proxy parameters
     public init(
         host: String,
         port: Int,
         token: String,
-        proxy: HTTPClient.Configuration.Proxy? = nil,
-        worker: HTTPClient.EventLoopGroupProvider
-    ) throws {
+				worker: HTTPClient.EventLoopGroupProvider,
```
Comment (givip) [inline Sources/Telegrammer/Network/BotClient.swift]: spaces instead tabs

**Review by givip — CHANGES_REQUESTED** [review]

**Review by givip — APPROVED** [review]

**Pull request merged**